packages/ra-core/src/controller/list/useList.spec.tsx

```diff
@@ -315,4 +315,47 @@ describe('<useList />', () => {
             );
         });
     });
+
+    describe('onSelectAll', () => {
+        it('should select all items if no items are selected', async () => {
+            const children = jest.fn();
+            const data = [{ id: 0 }, { id: 1 }];
+            render(<UseList data={data} callback={children} />);
+            act(() => {
+                children.mock.calls.at(-1)[0].onSelectAll();
+            });
+            await waitFor(() => {
+                expect(children).toHaveBeenCalledWith(
+                    expect.objectContaining({
+                        selectedIds: [0, 1],
+                    })
+                );
+            });
+        });
+        it('should select all items if some items are selected', async () => {
+            const children = jest.fn();
+            const data = [{ id: 0 }, { id: 1 }];
+            render(<UseList data={data} callback={children} />);
+            act(() => {
+                children.mock.calls.at(-1)[0].onSelect([1]);
+            });
+            await waitFor(() => {
+                expect(children).toHaveBeenCalledWith(
+                    expect.objectContaining({
+                        selectedIds: [1],
+                    })
+                );
+            });
+            act(() => {
+                children.mock.calls.at(-1)[0].onSelectAll();
+            });
+            await waitFor(() => {
+                expect(children).toHaveBeenCalledWith(
+                    expect.objectContaining({
+                        selectedIds: [0, 1],
+                    })
+                );
+            });
+        });
+    });
```

Comment:
missing test on the limit

Reply:
The `limit` prop is not usefull for the `useList` hook because this hook received all the data directly so we don't need to call it. (cf PR description > Informations)